pkg/provider/provider.go

```diff
@@ -64,6 +65,51 @@ func getNameFromComment(typeOfComment, comment string) string {
 	return strings.TrimSpace(getFirstLine[0])
 }
 
+func GetPipelineRunAndBranchNameFromTestComment(comment string) (string, string, error) {
+	if strings.Contains(comment, testComment) {
+		return getPipelineRunAndBranchNameFromComment(testComment, comment)
+	}
+	return getPipelineRunAndBranchNameFromComment(retestComment, comment)
+}
+
+func GetPipelineRunAndBranchNameFromCancelComment(comment string) (string, string, error) {
+	return getPipelineRunAndBranchNameFromComment(cancelComment, comment)
+}
+
+// getPipelineRunAndBranchNameFromComment function will take GitOps comment and split the comment
+// by /test, /retest or /cancel to return branch name and pipelinerun name.
+func getPipelineRunAndBranchNameFromComment(typeOfComment, comment string) (string, string, error) {
+	var prName, branchName string
+	splitTest := strings.Split(comment, typeOfComment)
+
+	// after the split get the second part of the typeOfComment (/test, /retest or /cancel)
+	// as second part can be branch name or pipelinerun name and branch name
+	// ex: /test branch:nightly, /test prname branch:nightly
+	if splitTest[1] != "" && strings.Contains(splitTest[1], ":") {
```

Comment:
what if the branch name has a : (that exist) woudl that work?

Reply:
As per my understanding we cannot create branch like `branch:test` 

I just tried by creating `branch:test` on CLI got below error

```
$ git checkout -b branch:test
fatal: 'branch:test' is not a valid branch name
```

Same thing tried on github UI

it created a branch `branchtest`

Reply:
ah that's interesting, that sounds good then